### DISCUSSION TITLE
Visionatrix: Simplifying and Scaling ComfyUI Workflows for Home and Production

### DISCUSSION THREAD

**Discussion by bigcat88** [comfyanonymous/ComfyUI#3569] [General]
Hello everyone,

We are excited to introduce Visionatrix( https://github.com/Visionatrix/Visionatrix ),- project aimed at enhancing the ComfyUI experience by simplifying its use and enabling seamless deployment and scaling of workflows both at home and in production environments.

### Key Advantages of Visionatrix over ComfyUI

1. **Easy Scaling:**
   - Visionatrix allows you to connect multiple servers to a single database and map them to a common folder where they write files. This setup simplifies the scaling process, making it easy to manage large-scale operations.

2. **Client-Consumer Mode:**
   - ComfyUI combines server and worker components. We separated these parts by providing a dedicated server that creates tasks and distributes them to workers upon request. This separation increases flexibility and efficiency.
   - **Advantages:** This allows better resource management, load balancing, and scalability. It ensures that the server can handle task distribution efficiently while workers focus on processing.

3. **Seamless Integration for Home Use:**
   - Despite the advanced features, Visionatrix is designed to be user-friendly. For home users, it operates as a unified system where both the server and worker parts function together seamlessly in a single process, masking the underlying complexities.

4. **Semi-Automatic Model Mapping:**
   - Visionatrix offers a solution for semi-automatic mapping of models, simplifying the configuration process and improving workflow efficiency.

5. **User Support and Authentication:**
   - Visionatrix includes support for multiple users and authentication, enhancing security and collaboration capabilities.

### Performance Testing

We tested Visionatrix on a basic VPS with 1 core and 1 GB of RAM, running a server with PostgreSQL and NGINX.
We connected up to three workers (NVIDIA 3060 12 Gb, AMD 7900 XTX, Apple M2 Max), and everything worked together smoothly, handling tasks in parallel without issues.

Since Vix supports all the command line options supported by ComfyUI and uses ComfyUI code to execute Nodes, the computing speed is the same or slightly faster(since there is no event loop for backend) as in ComfyUI.

### Discussion Points

We are interested in hearing from knowledgeable users:

- What are the potential drawbacks of this approach?
- How can we improve Visionatrix further?

Best regards, thank you in advance.

**Comment by andrey18106** [comment]
We have reworked [Visionatrix](https://github.com/Visionatrix/Visionatrix/releases/tag/v0.6.0) and ComfyUI workflows are now can be more [easily ported to Visionatrix](https://visionatrix.github.io/VixFlowsDocs/ComfyUI2VixMigration.html).

**Comment by bigcat88** [comment]
For those interested, we have released version **1.0**, which completely satisfies us in terms of functionality, and decided to move on to support `inpaint`, `magic prompt` and probably translations.

The main points:

1. No major breaking changes are expected in the project _(the only thing is that upcoming version `1.1` will include ComfyUI with PR 2666 - reversed flow execution)_
2. Added OpenAPI documentation
3. Added various useful nodes to the default installation since then
4. **ChildTasks**: Made it possible to send the results of one flow to another and for the result to be attached to the parent. That is, the user can apply both Background Removal, Scaling and other things to one image without going to other pages/workflows, and the results are always in front of him with possibility to roll back to the previous ones - it turned out to be a mini Photoshop =)

---

As this is our hobby project, we are always happy for any feedback, criticism, suggestions for improvement.

**Comment by bigcat88** [comment]
Last six months, the work continued.

Released version 2.0, switched to Comfy-UI manager (thanks to its author for quick merges of our PRs), made many improvements where possible.
There is still a lot of work and ideas for implementation ahead.

In the near future, we now plan to add automatic scaling using [Novita.ai](https://novita.ai) and [RunPOD.io](https://runpod.io), since even 24 gigabytes of cards are sometimes too little for flows.

I will probably close the discussion, if anyone is interested - we can always chat in the repository/discord. :)

![Screenshot from 2025-03-27 12-51-02](https://github.com/user-attachments/assets/186807a6-8daf-43ef-8b4a-63c3fec71aca)


**We wish everyone fast and high-quality workflows!**
